contracts/Wrapped1155Factory.sol

```diff
@@ -101,22 +104,29 @@ contract Wrapped1155Factory is ERC1155Receiver {
         uint256[] calldata tokenIds,
         uint256[] calldata amounts,
         address recipient,
+        bytes memory tokenName,
+        bytes memory tokenNameOffset,
         bytes calldata data
     )
         external
     {
         require(tokenIds.length == amounts.length, "Wrapped1155Factory: mismatched input arrays");
         for (uint i = 0; i < tokenIds.length; i++) {
-            getWrapped1155(multiToken, tokenIds[i]).burn(msg.sender, amounts[i]);
+            getWrapped1155(multiToken, tokenIds[i], tokenName, tokenNameOffset).burn(msg.sender, amounts[i]);
         }
         multiToken.safeBatchTransferFrom(address(this), recipient, tokenIds, amounts, data);
     }
 
-    function getWrapped1155DeployBytecode(IERC1155 multiToken, uint256 tokenId)
+    function getWrapped1155DeployBytecode(
+        IERC1155 multiToken, 
+        uint256 tokenId, 
+        bytes memory tokenName,
+        bytes memory tokenNameOffset
+    )
         public
         view
         returns (bytes memory)
-    {
+    {        
         return abi.encodePacked(
             // assign factory
             hex"73",
```

Comment:
I would explain these opcodes:
e.g. for `73` -> `// Pushes 20 bytes to memory, in this case the address of the factory`

For later opcodes:
`3d55`:
`3d` -> `returndatasize allows us to push a 0 onto stack with minimal gas costs`
`55`-> `Stores the data in stack storage 1 (factory address) at slot specified in stack storage 0 (0)`

`600255` (line 144):
`6002` -> `push the next 1 byte of the instruction data onto stack (02)`
`55`-> `Stores the data in stack storage 1 (tokenId) at slot specified in stack storage 0 (02)`

While assembly code is very nice I would be very explicit in documenting it

I know that this might be out of scope of this PR, but I feel this would be very useful for this repo

Reply:
Thanks for the samples @rmeissner , I aggre with that https://github.com/gnosis/1155-to-20/issues/5